### DISCUSSION TITLE
packages/test/framework.yml not read?

### DISCUSSION THREAD

**Discussion by sofigio** [symfony/symfony#42832] [Q&A] [answered]
**Symfony version(s) affected**: 5.3.7

I struggled with this issue for many hours, at the end I found a solution but I have reasons to believe it's a bug. 

Basically, I created a simple phpunit test:

    public function testPage()
    {
        $client = static::createClient();
        $client->request('GET', '/en/registration');
        $this->assertEquals(200, $client->getResponse()->getStatusCode());
    }

controller:

    public function registrationAction()
    {
        return $this->render('page.html.twig');
    }

Twig file:

    {% for label, messages in app.flashes %}flash!{% endfor %}

The test fails because the page returns 500 with:

> An exception has been thrown during the rendering of a template (&quot;Failed to start the session because headers have already been sent by &quot;...\vendor\bin\.phpunit\phpunit-9.5-0\src\Util\Printer.php&quot; at line 104.&quot;). (500 Internal Server Error)

The solution to this issue found on the web is to set `storage_id: session.storage.mock_file` in `packages/test/framework.yaml`... but I had it already! It looks like this file isn't read, it doesn't overwrite `packages/framework.yaml` anymore... If I put/uncomment this in packages/framework.yaml it works:

```
when@test:
    framework:
        test: true
        session:
            storage_factory_id: session.storage.factory.mock_file
```

Am I doing something wrong? From what I read [here](https://symfony.com/blog/new-in-symfony-5-3-configure-multiple-environments-in-a-single-file) the old system should continue to work...

**Comment by ro0NL** [comment]
Can you confirm your Kernel still tries to import this directory?

https://github.com/symfony/recipes/blob/f3b1d14e77717454f4f4c085e2bdd5ade9306d6f/symfony/framework-bundle/5.3/src/Kernel.php#L17

**Reply by sofigio** [reply]
yes, my Kernel.php is exactly the same as that one

**Reply by sofigio** [reply]
Other yaml files in the `packages/test/` directory works, 
for example if I add in doctrine.yaml:

```
doctrine:
    dbal:
        dbname: 'test123'
```

I got `SQLSTATE[HY000] [1049] Unknown database 'tm_test' (500 Internal Server Error)`

**Reply by sofigio** [reply]
I found the issue, is has nothing to do with importing yamls... sorry

**Comment by sofigio** [comment, marked as answer]
I found the issue, this is not working:

    storage_id: session.storage.mock_file

While this is working:

    storage_factory_id: session.storage.factory.mock_file

Even if the [changelog](https://github.com/symfony/framework-bundle/blob/5.3/CHANGELOG.md) says it's deprecated, not removed, in 5.3

**Reply by stof** [reply]
maybe you have another file that you already migrated to use `storage_factory_id`, meaning that the merged config contains both. If you configure both, the new `storage_factory_id` takes precedence.

**Reply by sofigio** [reply]
you're right, I have `storage_factory_id` in main framework.yaml... didn't notice this. Thank you
